src/neural/cuda/network_cuda.cc

```diff
@@ -490,7 +516,17 @@ class CudaNetwork : public Network {
     }
 
     // Policy head.
-    if (conv_policy_) {
+    if (attn_policy_) {
+      network_[l++]->Eval(batchSize, tensor_mem[0], tensor_mem[2],
```

Comment:
The attention policy head assumes these inputs of tensor_mem are at least as big as scratch_mem.  But the logic only currently guarantees that if res block fusing is enabled.  That check should be changed to also include attn_policy_.  Not sure but maybe also need to ensure that size is at least twice as big as the outputs of this layer...

Reply:
added some checks when allocating these allocations. Hopefully should work for most networks otherwise the backend will exit with error.